### PR TITLE
feature: Connect posts fully with backend

### PR DESCRIPTION
**Pull request by JimTheCat**

### CODE REVIEW

**Comment by JimTheCat** [comment]
Task linked: [CU-869768m7p Connect PostsMenu with backend](https://app.clickup.com/t/869768m7p)

**Comment by JimTheCat** [comment]
I will additionally create a Infinite Scroll for Main page

File: frontend/src/Features/MainPage/components/InfiniteScroll/InfiniteScroll.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      {/* Loading indicator */}
+      {loading && (
+        <Box style={{textAlign: "center", margin: "20px 0"}}>
+          <Loader size="md"/>
+        </Box>
+      )}
+      {/* End of data */}
+      {!hasMore && !loading && (
+        <Text ta="center" c="dimmed" mt="lg">
+          No more posts ಥ_ಥ
```
Comment (KinTrae) [inline frontend/src/Features/MainPage/components/InfiniteScroll/InfiniteScroll.tsx:60]: XD nice one

**Review by KinTrae — APPROVED** [review]
Looks great😎

**Pull request merged**